### ISSUE TITLE
Copy Construct warning

### ISSUE THREAD

**Issue by vince62s** [pytorch/text#467]
Hello,
was this ever fixed ?

/usr/local/lib/python3.5/dist-packages/torchtext/data/field.py:308: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).

**Comment by vince62s** [comment]
it refers to this line in master:
https://github.com/pytorch/text/blob/master/torchtext/data/field.py#L359

**Comment by toli-belo** [comment]
also what if you need to convert tensor to dtype=torch.float ? 
replacing with clone().detach() is not going to convert the tensor....

**Comment by mttk** [comment]
Could you provide the case where you get this error?

I assume that `arr` is a `torch.tensor`, which means that the numericalization / postprocessing casts it into that at some point (as the standard pipeline shouldn't do that prior to [L359](https://github.com/pytorch/text/blob/master/torchtext/data/field.py#L359)).

**Comment by zhangguanheng66** [comment]
@toli-belo feel free to reopen the issue if you still have questions.

**Comment by martijnvanattekum** [comment]
for future reference: https://github.com/pytorch/pytorch/issues/16627. In summary, try to replace `torch.tensor` with `torch.as_tensor`

**Comment by meetshrihari** [comment]
Replacing 'torch.tensor' with 'torch.as_tensor' removed the warning.
Thank you.
